SDLSecurity/_SDLCertificateManager.m

```diff
@@ -83,80 +85,58 @@ - (nullable NSData *)certificateData {
         return nil;
     }
     
-    // TODO: Decrypt?
-    
-    // TODO: Return data
     return [NSData dataWithContentsOfFile:certPath];
 }
 
 - (void)retrieveNewCertificateWithAppId:(NSString *)appId completionHandler:(SDLCertificateRetrievedHandler)completionHandler {
-    SDLSecurityLogD(@"Performing network request for certificate");
+    SDLSecurityLogD(@"Performing network request for a certificate");
 
-    // Create the URL request with the required parameters
-    NSMutableURLRequest *request = [[NSURLRequest requestWithURL:self.certificateURL] mutableCopy];
-    if (request == nil) {
-        SDLSecurityLogE(@"Error creating security request from URL");
-        completionHandler(NO, nil); // TODO: Error
-        return;
-    }
-    
-    [request setValue:@"application/json" forHTTPHeaderField:@"content-type"];
-    [request setValue:@"application/json" forHTTPHeaderField:@"accept"];
-    
-    // Attempt to serialize the app id dictionary into json data
-    NSError *jsonEncodeError = nil;
-    NSDictionary *appIdJSON = @{@"appId": appId};
-    NSData *appIdData = [NSJSONSerialization dataWithJSONObject:appIdJSON options:0 error:&jsonEncodeError];
-    if (appIdData == nil) {
-        SDLSecurityLogE(@"Error creating security request JSON");
-        completionHandler(NO, jsonEncodeError);
-        return;
-    }
+    NSArray<NSURLQueryItem *> *queryItems = @[[[NSURLQueryItem alloc] initWithName:@"appID" value:appId]];
+    NSURLComponents *queryComponents = [NSURLComponents componentsWithString:self.certificateURL];
+    queryComponents.queryItems = queryItems;
+    NSURL *url = queryComponents.URL;
 
-    [[NSURLSession sharedSession] uploadTaskWithRequest:request fromData:appIdData completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
-        // If the certificate server didn't send us any data back
-        if (data == nil) {
-            SDLSecurityLogE(@"Security server responded with an error. Response: %@, Error: %@", response, error);
-            completionHandler(NO, error);
-            return;
-        }
+    NSURLSessionConfiguration *sessionConfig = [NSURLSessionConfiguration defaultSessionConfiguration];
+    sessionConfig.timeoutIntervalForRequest = 20.0;
+
+    NSURLSession *session = [NSURLSession sharedSession];
+    __weak typeof(self) weakSelf = self;
+    NSURLSessionDataTask *task = [session dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
```

Comment:
The task should probably be stored so that it can be canceled?

Reply:
I created issue #30 to address this comment.